synthesis/codon/codon.go

```diff
@@ -85,39 +99,131 @@ type AminoAcid struct {
 
 // Table is an interface that specifies the functions that all table types must implement
 type Table interface {
-	Chooser() (map[string]weightedRand.Chooser, error)
-	GenerateTranslationTable() map[string]string
-	GenerateStartCodonTable() map[string]string
-	GetAminoAcids() []AminoAcid
-	GetStartCodons() []string
-	GetStopCodons() []string
-	IsEmpty() bool
-	OptimizeTable(string) Table
+	GetWeightedAminoAcids() []AminoAcid
+	OptimizeSequence(aminoAcids string, randomState ...int) (string, error)
 }
 
-// codonTable holds information for a codon table.
-type codonTable struct {
+// TranslationTable holds information for a codon table.
+type TranslationTable struct {
```

Comment:
There is probably a better name for this struct, but since the name stuttered I went for this. I'll try to think of a better one!

This struct now does both Translation and Optimization. Previously `Translation` and `OptimizeSequence` would re-compute the `TranslationMap` and `StartCodonTable` or `Chooser` maps every time through the `Table` interface. Those are now computed in the struct's constructor's and upon the weights being updated with a new sequence.

Reply:
Eh, close enough! I like it.